at_client/src/main/java/org/atsign/client/util/RegisterUtil.java

```diff
@@ -0,0 +1,108 @@
+package org.atsign.client.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.atsign.common.AtException;
+import org.json.JSONObject;
+
+public class RegisterUtil {
+    //Calls the API to receive atsigns that are ready to be claimed. Returns a free atsign.
+    public String getFreeAtsign() throws AtException, MalformedURLException, IOException {
+        URL urlObject = new URL(Constants.AT_DEV_DOMAIN + Constants.API_PATH + Constants.GET_FREE_ATSIGN);
+        HttpsURLConnection connection = (HttpsURLConnection) urlObject.openConnection();
```

Comment:
Interesting. I would expect the setters to be called before calling the connect method.